src/mono/mono/metadata/class-init.c

```diff
@@ -816,7 +816,7 @@ has_inline_array_attribute_value_func (MonoImage *image, uint32_t method_token,
 		MonoDecodeCustomAttr *decoded_attr = mono_reflection_create_custom_attr_data_args_noalloc (image, ctor, (guchar*)data, data_size, &error);
 		mono_error_assert_ok (&error);
 		g_assert (decoded_attr->named_args_num == 0 && decoded_attr->typed_args_num == 1);
-		attr->value = *(gpointer*)decoded_attr->typed_args [0]->value.primitive;
+		attr->value = (gpointer)*(guint32*)decoded_attr->typed_args [0]->value.primitive;
```

Comment:
```suggestion
		attr->value = GUINT_TO_POINTER (read32 (decoded_attr->typed_args [0]->value.primitive));
```

(you may need to include `"mono/metadata/mono-endian.h"` for `read32`)